lib/mongoid/contextual/atomic.rb

```diff
@@ -163,11 +175,97 @@ def unset(*args)
         view.update_many("$unset" => Hash[fields])
       end
 
+      # Performs an atomic $currentDate operation on the given field or fields.
+      #
+      # @example Set the "last viewed" timestamp.
+      #   context.current_date(last_viewed: :timestamp)
+      #
+      # @example Set the completion date.
+      #   context.current_date(:completed)
+      #   # or
+      #   context.current_date(completed: true)
+      #
+      # @example Set multiple fields to the current date.
+      #   context.current_date(:completed, last_viewed: :timestamp)
+      #
+      # @param [ Array<Hash | Symbol | String> ] fields The fields with the
+      #   corresponding date format to use (true, :timestamp, or :date)
+      #
+      # @return [ nil ] Nil.
+      def current_date(*fields)
+        fields = collect_nested_operations(fields) { |value| translate_date_type(value) }
+        view.update_many("$currentDate" => fields)
+      end
+
+      # Performs an atomic $min update operation on the given field or fields.
+      # Each field will be set to the minimum of [current_value, given value].
+      # This has the effect of making sure that each field is no
+      # larger than the given value; in other words, the given value is the
+      # effective *maximum* for that field.
+      #
+      # @note Because of the existence of
+      #   Mongoid::Contextual::Aggregable::Mongo#min, this method cannot be
+      #   named #min, and thus breaks that convention of other similar methods
+      #   of being named for the MongoDB operation they perform.
+      #
+      # @example Set "views" to be no more than 100.
+      #   context.update_min(views: 100)
+      #
+      # @param [ Hash ] fields The fields with the maximum value that each
+      #   may be set to.
+      #
+      # @return [ nil ] Nil.
+      def update_min(fields)
```

Comment:
Using the word `update_` here is confusing vis-a-vis `update_attributes` in Mongoid, which invokes model callbacks. As far as Mongoid is concerned, we should reserve the word `update` exclusively for actions which do callbacks.

(Yes the driver has `update_many`, but Mongoid users will not ever be using these driver methods directly.)

I would propose something like `atomic_min` or `set_to_min` as an alternative (the latter being congruent with the existing atomic `set` method.)

Reply:
Good point about the `update` prefix; I'll switch to `set`, which I agree is clearer. I agree that the `min`/`max` naming is confusing, but I think it is important that the canonical method mirror the naming of the underlying MongoDB operation. That said, I'm totally on board with having an alias or two to make the intention clearer.

Reply:
usually I'm not a big fan of aliases but in this particular case `min/max` are going to be impossible for humans to member which is which here.